subprojects/core/src/main/java/org/gradle/internal/service/scopes/DefaultGradleUserHomeScopeServiceRegistry.java

```diff
@@ -115,27 +118,23 @@ public Optional<ServiceRegistry> getCurrentServices() {
 
     @Override
     public void release(ServiceRegistry registry) {
-        lock.lock();
```

Comment:
what if this throws an error, but still manage to lock? Should care/catch this as well?